src/main/webapp/app/tasks/local-task.service.ts

```diff
@@ -32,12 +32,32 @@ export class LocalTaskService implements TaskService {
     return of(null);
   }
 
+  update(task: Task): Observable<void> {
+    const tasks = this.readTasks();
+    const index = tasks.findIndex(taskItem => taskItem.id === task.id);
+    if (index !== -1) {
+      tasks.splice(index, 1, task);
+      this.writeTasks(tasks);
+      this.getAll();
+    }
+    return of(null);
```

Comment:
From an API design viewpoint it could make sense to return the edited task as an `Observable< Task>`